aws/ec2_test.go

```diff
@@ -0,0 +1,132 @@
+package aws
+
+import (
+	"bytes"
+	"math/rand"
+	"testing"
+	"time"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+var uniqueTestID = "aws-nuke-test-" + uniqueID()
+
+// Returns a unique (ish) id we can attach to resources and tfstate files so they don't conflict with each other
+// Uses base 62 to generate a 6 character string that's unlikely to collide with the handful of tests we run in
+// parallel. Based on code here: http://stackoverflow.com/a/9543797/483528
+func uniqueID() string {
+
+	const BASE_62_CHARS = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+	const UNIQUE_ID_LENGTH = 6 // Should be good for 62^6 = 56+ billion combinations
+
+	var out bytes.Buffer
+
+	rand := rand.New(rand.NewSource(time.Now().UnixNano()))
+	for i := 0; i < UNIQUE_ID_LENGTH; i++ {
+		out.WriteByte(BASE_62_CHARS[rand.Intn(len(BASE_62_CHARS))])
+	}
+
+	return out.String()
+}
+
+func createTestEC2Instance(t *testing.T, session *session.Session) ec2.Instance {
+	svc := ec2.New(session)
+
+	params := &ec2.RunInstancesInput{
+		ImageId:      awsgo.String("ami-e7527ed7"),
+		InstanceType: awsgo.String("t1.micro"),
+		MinCount:     awsgo.Int64(1),
+		MaxCount:     awsgo.Int64(1),
+	}
+
+	runResult, err := svc.RunInstances(params)
+	if err != nil {
+		assert.Failf(t, "Could not create test EC2 instance: %s", errors.WithStackTrace(err).Error())
+	}
+
+	// Add test tag to the created instance
+	_, err = svc.CreateTags(&ec2.CreateTagsInput{
+		Resources: []*string{runResult.Instances[0].InstanceId},
```

Comment:
I wonder if, due to AWS eventual consistency, this API call can fail, as the information about this instance ID has not yet propagated... You may need to call [WaitUntilInstanceExists](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.WaitUntilInstanceExists).

Reply:
Done

Reply:
Done